apps/wallet-mobile/src/features/Receive/common/ShareDetailsCard/useLastDateAddressUsed.ts

```diff
@@ -0,0 +1,28 @@
+import {useMemo} from 'react'
+import {useIntl} from 'react-intl'
+
+import {useSelectedWallet} from '../../../../SelectedWallet'
+
+export const useLastDateAddressUsed = (address: string) => {
+  const wallet = useSelectedWallet()
+  const intl = useIntl()
+  const lastUsed = useMemo(
+    () =>
+      Object.values(wallet.transactions).reduce((currentLast, tx) => {
+        const {inputs, outputs} = tx
+        const isRelevant = inputs.some((v) => address === v.address) || outputs.some((v) => address === v.address)
+        if (!isRelevant) return currentLast
+        const lastUpdatedAt = new Date(tx.lastUpdatedAt).getTime()
+
+        return Math.max(currentLast, lastUpdatedAt)
+      }, 0),
+    [address, wallet.transactions],
+  )
+  return lastUsed
+    ? intl.formatDate(new Date(lastUsed), {
+        dateStyle: 'short',
+        timeStyle: 'short',
+        hour12: false,
+      })
+    : null
```

Comment:
```suggestion
  return React.useMemo(
    () =>
      Object.values(wallet.transactions).reduce((currentLast, tx) => {
        const {inputs, outputs} = tx
        const isRelevant = inputs.some((v) => address === v.address) || outputs.some((v) => address === v.address)
        if (!isRelevant) return currentLast
        const lastUpdatedAt = new Date(tx.lastUpdatedAt).getTime()

        const lastUsed = Math.max(currentLast, lastUpdatedAt)
        const formattedLastUsed = lastUsed
    ? intl.formatDate(new Date(lastUsed), {
        dateStyle: 'short',
        timeStyle: 'short',
        hour12: false,
      })
    : null
      }, 0),
    [address, wallet.transactions],
  )
```

Reply:
I don't think that would work, the reducer uses a number as accumulator, not a formatted date.

Reply:
If the intention is to just memo the number to formatted as well, I can move that inside the memo, but not inside the reducer.

Reply:
it was a suggestion, to collocate the logic, and don't kick the hook entirely since there will be only hooks in the body, it needs to adjust the memo function `{..yayada, return..}`